lib/ocpp/common/websocket/websocket_libwebsockets.cpp

```diff
@@ -449,20 +449,22 @@ void WebsocketTlsTPM::client_loop() {
 
         if (this->connection_options.security_profile == 3) {
 
-            const auto certificate_key_pair =
-                this->evse_security->get_key_pair(CertificateSigningUseEnum::ChargingStationCertificate);
+            const auto certificate_info =
+                this->evse_security->get_leaf_certificate_info(CertificateSigningUseEnum::ChargingStationCertificate);
 
-            if (!certificate_key_pair.has_value()) {
+            if (!certificate_info.has_value()) {
                 EVLOG_AND_THROW(std::runtime_error(
                     "Connecting with security profile 3 but no client side certificate is present or valid"));
             }
 
-            path_chain = certificate_key_pair.value().certificate_path;
-            if (path_chain.empty()) {
-                path_chain = certificate_key_pair.value().certificate_single_path;
+            if (certificate_info.value().certificate_path.has_value()) {
+                path_chain = certificate_info.value().certificate_path.value();
+            } else {
+                path_chain = certificate_info.value().certificate_single_path.value();
```

Comment:
This is an unchecked optional access afaics

Reply:
Theoretically, if the response is 'Accepted' at least one of those values should be valid. If they are not, it is an exception that should crash the system, and see the root of the cause, why both are invalid.

Reply:
Yep, but this could change in the future and I would rather throw a proper exception than getting a bad optional access here.